lib/src/js/value/color.dart

```diff
@@ -72,7 +372,68 @@ final JSClass colorClass = () {
 ///
 /// This ensures that an explicitly null alpha will be treated as a missing
 /// component.
-num? _handleUndefinedAlpha(num? alpha) => isUndefined(alpha) ? 1 : alpha;
+double? _handleUndefinedAlpha(double? alpha) => isUndefined(alpha) ? 1 : alpha;
+
+/// This procedure takes a `channel` name, an object `changes` and a SassColor
+/// `initial` and returns the result of applying the change for `channel` to
+/// `initial`.
+double? _changeComponentValue(
+    SassColor initial, String channel, Map<String, double?> changes) {
+  var initialValue = initial.channel(channel);
+  if (!changes.containsKey(channel)) {
+    return initialValue;
+  }
+  return changes[channel];
+}
+
+/// Determines the construction space based on the provided options.
+ColorSpace _constructionSpace(_ConstructionOptions options) {
+  if (options.space != null) return ColorSpace.fromName(options.space!);
+  if (options.red != null) return ColorSpace.rgb;
+  if (options.saturation != null) return ColorSpace.hsl;
+  if (options.whiteness != null) return ColorSpace.hwb;
+  throw "No color space found";
+}
+
+void _checkNullAlphaDeprecation(_ConstructionOptions options) {
+  // @todo Verify if options.space is equivalent to "not set"
+  if (options.alpha == null && options.space == null) {
+    _emitNullAlphaDeprecation();
+  }
+}
+
+void _emitNullAlphaDeprecation() {
+  warnForDeprecationFromJsApi(
+      "Passing `alpha: null` without setting `space` is deprecated."
+      "\n"
+      "More info: https://sass-lang.com/d/null-alpha",
+      Deprecation.nullAlpha);
+}
+
+void _emitColor4ApiDeprecation(String name) {
+  warnForDeprecationFromJsApi(
+      "$name is deprecated, use `channel` instead.", Deprecation.color4Api);
+}
+
+Map<String, double?> _changedOptions(_ConstructionOptions options) {
```

Comment:
You can do this more easily with [`jsify()`](https://api.dart.dev/stable/3.1.3/dart-js_util/jsify.html). But why bother converting this to a map in the first place instead of just interacting with it as a JS Object?